app/src/main/java/com/example/drawingApp/DrawingFieldView.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.drawingApp
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+
+@SuppressLint("ClickableViewAccessibility")
+/**
+ * Creates a Field to draw in with a color
+ * use update color to change the stroke color
+ * uses an internal canvas to draw on a saved bitmap member variable
+ */
+class DrawingFieldView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    private var paint = Paint().apply {
+        color = Color.BLACK
+        style = Paint.Style.STROKE
+        strokeWidth = 4f
+        //avoid the issue of dealing handling possible uninitialized values, black is expected on starting a drawing app after all
+    }
+    private var canvasBitMap: Bitmap? = null
+    private var held = false    //used to track if the user is holding finger down on the view
```

Comment:
You could get rid of the comment if you made this "isTouchHeldDown" or something similar

Reply:
named it heldDown, for a shorter form but similar idea